### PR TITLE
type overhaul

### PR DESCRIPTION
**Pull request by megagreg72**
It seems like `zod` is introducing code duplication atm, so we can either remove it and validate ourselves by writing functions like this, or just use `zod` better.
This is how I typically define sets of strings (essentially enums) without a library like `zod`.
I'll give it a try to see if it helps.

### CODE REVIEW

File: src/shared/types/sets/movement.ts [hunk 1]
```diff
@@ -0,0 +1,18 @@
...
+  "pipe",
+] as const;
+
+export type MovementType = (typeof movementTypes)[number];
+
+export function isMovementType(x: any): x is MovementType {
+  return (
+    typeof x === "string" && (movementTypes as readonly string[]).includes(x)
+  );
+}
```
Comment (megagreg72) [inline src/shared/types/sets/movement.ts:18]: we might not even need zod if we use functions like this for runtime validation
but idk, I don't fully understand zod yet

File: src/shared/types/sets/movement.ts [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+  "tires",
+  "air",
+  "sea",
+  "lander",
+  "pipe",
+] as const;
+
+export type MovementType = (typeof movementTypes)[number];
+
+export function isMovementType(x: any): x is MovementType {
```
Comment (FunctionDJ) [inline src/shared/types/sets/movement.ts:14]: we never need to validate a movement type. it's not a user input.

**Review by FunctionDJ — COMMENTED** [review]
the purpose of zod is to create types and validation with a single structure: a schema.
if we remove zod, we need more code: some for the type, and then some for the validation. and they won't be kept in sync by default.

**Comment by megagreg72** [comment]
after messing around a bit more, I think `zod` is useful but we could use it a bit better. I'll close this PR

**Pull request closed without merging**